### ISSUE TITLE
Release GIFT artifacts (models, datasets) on Hugging Face

### ISSUE THREAD

**Issue by NielsRogge** [MIT-CLEAR-Lab/GIFT#1]
Hi @FinAminToastCrunch 🤗

Niels here from the open-source team at Hugging Face. I discovered your work on Arxiv (GIFT: Generalizing Intent for Flexible Test-Time Rewards) and was wondering whether you would like to submit it to [hf.co/papers](https://hf.co/papers) to improve its discoverability. If you are one of the authors, you can submit it at https://huggingface.co/papers/submit.

The paper page lets people discuss about your paper and lets them find artifacts about it (your models, datasets or demo for instance), you can also claim the paper as yours which will show up on your public profile at HF, add Github and project page URLs.

I noticed your Github repository is currently being prepared. It'd be great to make the reward function checkpoints and the robot demonstration datasets available on the 🤗 hub when you are ready to release them, to improve their discoverability and visibility within the robotics community.

## Uploading models

See here for a guide: https://huggingface.co/docs/hub/models-uploading.

In this case, for your learned reward functions or intent-conditioned similarity components, we could leverage the [PyTorchModelHubMixin](https://huggingface.co/docs/huggingface_hub/package_reference/mixins#huggingface_hub.PyTorchModelHubMixin) class which adds `from_pretrained` and `push_to_hub` to any custom `nn.Module`.

## Uploading datasets

Would be awesome to make the robot demonstration datasets or evaluation benchmarks available on 🤗, so that people can do:

```python
from datasets import load_dataset

dataset = load_dataset("your-hf-org-or-username/your-dataset")
```
See here for a guide: https://huggingface.co/docs/datasets/loading.

Besides that, there's the [dataset viewer](https://huggingface.co/docs/hub/en/datasets-viewer) which allows people to quickly explore the data in the browser.

Let me know if you're interested/need any help regarding this!

Cheers,

Niels
ML Engineer @ HF 🤗
